From a31c06501e20595b2d8509a1b0d81eafda576131 Mon Sep 17 00:00:00 2001
From: Carlo Caione <ccaione@baylibre.com>
Date: Thu, 15 Nov 2018 11:12:17 +0000
Subject: [PATCH 35/55] mender: apollo: Introduce the initramfs

The standard mender script doesn't support passing an initramfs
argument. Fix that.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>
---
 include/config_mender_defines.h | 4 +---
 include/env_mender.h            | 9 ++++++++-
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/include/config_mender_defines.h b/include/config_mender_defines.h
index 74180f3e5f..f728b24aac 100644
--- a/include/config_mender_defines.h
+++ b/include/config_mender_defines.h
@@ -1,6 +1,3 @@
-/* AUTOGENERATED FILE - DO NOT EDIT! */
-/* This file is provided by the meta-mender layer. */
-
 #ifndef HEADER_CONFIG_MENDER_DEFINES_H
 #define HEADER_CONFIG_MENDER_DEFINES_H
 
@@ -26,6 +23,7 @@
 
 #define MENDER_BOOT_KERNEL_TYPE "booti"
 #define MENDER_KERNEL_NAME "Image"
+#define MENDER_RAMDISK_NAME "apollo-initramfs-image-apollo-evt0.cpio.gz"
 #define MENDER_DTB_NAME "meson-axg-apollo-evt0.dtb"
 #define MENDER_UBOOT_PRE_SETUP_COMMANDS ""
 #define MENDER_UBOOT_POST_SETUP_COMMANDS ""
diff --git a/include/env_mender.h b/include/env_mender.h
index 6fde590e5c..644e0dcdf3 100644
--- a/include/env_mender.h
+++ b/include/env_mender.h
@@ -63,6 +63,8 @@
                                                                         \
     "mender_kernel_name=" MENDER_KERNEL_NAME "\0"                       \
                                                                         \
+    "mender_ramdisk_name=" MENDER_RAMDISK_NAME "\0"                     \
+                                                                        \
     "mender_dtb_name=" MENDER_DTB_NAME "\0"                             \
                                                                         \
     "mender_pre_setup_commands=" MENDER_UBOOT_PRE_SETUP_COMMANDS "\0"   \
@@ -138,13 +140,18 @@
     "load ${mender_uboot_root} ${fdt_addr_r} /boot/${mender_dtb_name}; " \
     "fi; "                                                              \
     "load ${mender_uboot_root} ${kernel_addr_r} /boot/${mender_kernel_name}; "
+# define MENDER_LOAD_RAMDISK                                            \
+    "if test \"${ramdisk_addr_r}\" != \"\"; then "                      \
+    "load ${mender_uboot_root} ${ramdisk_addr_r} /boot/${mender_ramdisk_name}; " \
+    "fi; "
 #endif
 
 #define CONFIG_MENDER_BOOTCOMMAND                                       \
     "run mender_setup; "                                                \
     MENDER_BOOTARGS                                                     \
     MENDER_LOAD_KERNEL_AND_FDT                                          \
-    "${mender_boot_kernel_type} ${kernel_addr_r} - ${fdt_addr_r}; "     \
+    MENDER_LOAD_RAMDISK                                                 \
+    "${mender_boot_kernel_type} ${kernel_addr_r} ${ramdisk_addr_r}:${filesize} ${fdt_addr_r}; " \
     "run mender_try_to_recover"
 
 #endif /* !MENDER_AUTO_PROBING */
-- 
2.19.1

